qemu/qemu.go

```diff
@@ -2553,8 +2550,13 @@ func (config *Config) appendMemoryKnobs() {
 	config.qemuParams = append(config.qemuParams, "-object")
 	config.qemuParams = append(config.qemuParams, objMemParam)
 
-	config.qemuParams = append(config.qemuParams, "-numa")
-	config.qemuParams = append(config.qemuParams, numaMemParam)
+	if isDimmSupported(config) {
+		config.qemuParams = append(config.qemuParams, "-numa")
+		config.qemuParams = append(config.qemuParams, numaMemParam)
+	} else {
+		config.qemuParams = append(config.qemuParams, "-machine")
```

Comment:
is this correct? `memory-backend` can be a machine type ?? I don't see it in x86

Reply:
`memory-backend` is not a machine type like `pc`, `q35`, or `s390-ccw-virtio`, but it can be used as a key in `-machine` (like `accel=kvm`) (see [QEMU doc](https://qemu-project.gitlab.io/qemu/system/invocation.html?highlight=memory-backend#hxtool-0)). I am adding `-machine` again because it is possible to add such specs even when `-machine` has already been specced and it is easier to add it later rather than modify the construction of the `-machine` flag earlier (e.g. `qemu-system-s390x -machine s390-ccw-virtio... [all sorts of other flags] -machine memory-backend=dimm1`).